apps/contribution/test/contribution.js

```diff
@@ -0,0 +1,349 @@
+        'sender does not have permission to veto';
+      });
+    });
+
+    it("should revert when veto contribution that does not exist", async () => {
+      const contributionId = await contribution.contributionsCount();
+      return assertRevert(async () => {
+        await contribution.veto(contributionId.toNumber()+1, {from: root});
+        'contribution not found';
+      });
+    });
+
+    it("veto contribution", async () => {
+      const contributionId = await contribution.contributionsCount();
+      let contributionObject = await contribution.getContribution(contributionId.toNumber());
+      console.log("veto block: " + contributionObject[7]);
+      console.log("current block: " + await getBlockNumber());
+
+      await contribution.veto(contributionId.toNumber(), {from: root});
+      // eslint-disable-next-line no-undef
+      assert(contributionObject[9], true);
+    });
+  });
+
+  describe("Claim contribution", async () => {
+    let contributionId;
+
+    // eslint-disable-next-line no-undef
+    before(async () =>{
+      let amount = 200;
+      let contributorId = 1;
+      let hashDigest = '0x0000000000000000000000000000000000000000000000000000000000000000';
+      let hashFunction = 1;
+      let hashSize = 1;
+  
+      await contribution.add(amount, contributorId, hashDigest, hashFunction, hashSize, {from: root});
+    
+      contributionId = await contribution.contributionsCount();
+    });
+
+    it("should revert when claim contribution that does not exist", async () => {
+      return assertRevert(async () => {
+        await contribution.claim(contributionId.toNumber()+1, {from: root});
+        'contribution not found';
+      });
+    });
+
+    it("should revert when claim contribution before confirmation block", async () => {
+      return assertRevert(async () => {
+        await contribution.claim(contributionId.toNumber(), {from: root});
+        'contribution not confirmed yet';
+      });
+    });
+
+    it("claim contribution", async () => {
+      if(contributionId < 10) {
+        await timeTravel(100);
+      }
+      else {
+        await timeTravel(blocksToWait);
+      }
+      await mineBlock();
+
+      let contributionObject = await contribution.getContribution(contributionId.toNumber());
+      console.log("claim block: " + contributionObject[7]);
+      console.log("current block: " + await getBlockNumber());
+
+      await contribution.claim(contributionId);
+      // eslint-disable-next-line no-undef
+      assert(contributionObject[3], true);
+    });
+
+    it("should revert when claim already claimed contribution", async () => {
+      return assertRevert(async () => {
+        await contribution.claim(contributionId.toNumber(), {from: root});
+        'contribution already claimed';
+      });
+    });
+
+  });
+
+  describe("Veto claimed contribution", async () => {
+    // eslint-disable-next-line no-undef
```

Comment:
idk if this test is really necessary as if a contribution is claimed, so theoretically it is impossible to veto because of block number @bumi